.github/workflows/rimage.yml

```diff
@@ -0,0 +1,60 @@
+---
+# SPDX-License-Identifier: BSD-3-Clause
+# Tools that can save round-trips to github and a lot of time:
+#
+# yamllint -f parsable this.yml
+# pip3 install ruamel.yaml.cmd
+# yaml merge-expand this.yml exp.yml && diff -w -u this.yml exp.yml
+#
+# github.com also has a powerful web editor that can be used without
+# committing.
+
+name: rimage
+
+# yamllint disable-line rule:truthy
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - tools/rimage/**
+  push:
+    paths:
+      - tools/rimage/**
+
+jobs:
+
+  # Basic build test
+  build:
+    runs-on: ubuntu-22.04
+    env:
+      # FIXME: add  -Wpointer-arith
+      _CFLGS: -Werror -Wall -Wmissing-prototypes
+        -Wimplicit-fallthrough=3
+
```

Comment:
That would be great but I have no idea what it would take and I don't have any Windows system right now.

I can help with the Github Actions aspects if you want to try to add it later. For now this PR is merely trying to restore lost functionality.

Reply:
Here is recipe for windows:
```
export OPENSSL_PATH="c:/msys64/var/lib/pacman/local/openssl-1.1.1.n-1"
export CAT_PATH="c:/Program Files/Git/usr/bin"
export NINJA_PATH="c:/Program/ Files/ninja"
export MSYS_INSTALL_DIR="C:/msys64/usr/bin/"
export CMAKE_GENERATOR=Ninja

cmake -B zephyrproject/build-rimage -S zephyrproject/sof/rimage
cmake --build zephyrproject/build-rimage
```
Some of us are windows users and it would be nice if building would be checked for them...
Maybe open issue with that and leave it for future development?

Reply:
> Some of us are windows users and it would be nice if building would be checked for them...

Absolutely yes of course. But let's please first restore previous functionality and make sure rimage compiles without warning on ANYTHING :-)

> export OPENSSL_PATH="c:/msys64/var/lib/pacman/local/openssl-1.1.1.n-1"
> ... 

I doubt this will all work with Github's Windows image but we can probably re-use some of df1ba222306daa0

> Maybe open issue with that and leave it for future development?

Depends on how long it takes, it could be quick (famous last words) and not really require a new issue.

Reply:
> I doubt this will all work with Github's Windows image but we can probably re-use some of https://github.com/thesofproject/sof/commit/df1ba222306daa02d9b49f69983cc69eaf7fa78a

Wait: this commit _already_ compiles rimage on Windows (without -Werr -Wall). It's a bit lost in the long build logs but it's there. So unless you recommend a different toolchain (which one?) with a different set of warnings maybe  then adding Windows here too wouldn't give anything new?

Reply:
unless it will not change in the future then I don't see any problems.